src/server/state.ts

```diff
@@ -71,8 +78,19 @@ export class Player extends Schema implements IPlayer {
   @type('int8')
   lives: number = DEFAULT_LIVES;
 
+  @type('int16')
+  score: number = 0;
+
+  @type('int32')
+  color: number = 0xffffff;
+
   @type('boolean')
   hasWon: boolean = false;
+
+  public addScore(deltaScore: number): void {
+    this.score += deltaScore;
+    if (this.score < 0) this.score = 0;
```

Comment:
Do we really want to put this check? Having a negative score could be a thing. If I look at games like Smash, they give negative scores.